fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/serialization/LoanApplicationValidator.java

```diff
@@ -1410,7 +1431,10 @@ public void validateForModify(final JsonCommand command, final Loan loan) {
 
             // validate if disbursement date is a holiday or a non-working day
             validateDisbursementDateIsOnNonWorkingDay(expectedDisbursementDate);
-            Long officeId = client != null ? client.getOffice().getId() : group.getOffice().getId();
+            final Long officeId = setOfficeId(client, group);
+            if (officeId == null) {
```

Comment:
Same as above.

Reply:
The reason to add this return  there, is because previously we were generating a HTTP 400 Bad Request when some required field like expectedDisbursementDate or principal or transactionProcessingStrategy, etc are missed. 

Now with the new way, we have those schema validations and other business logic mixed, then the business validations are returning first than schema validations.